### PR TITLE
feat: ability for reviewers to give scores

### PR DESCRIPTION
**Pull request by owengretzinger**
Added the ability for reviewers to give scores:

- Created tRPC procedure for submitting a score
- Updated frontend to include review panel
  - For reviewers who have not reviewed that application yet, it shows a form for entering a score & comment (includes error handling for invalid score)
  - For admins, and reviewers who have already reviewed that application, it shows the average score, then each individual score with comments.
- Added review count column to table
- Made it so that the UpdateStatusDropdown can only be used by admins
- Modified tRPC procedure for reviewer.getApplications to also include the number of reviews (to display it in the table)

REVIEWER:

https://github.com/user-attachments/assets/cdf992be-06a3-4332-839a-01c6fa63a85a

ADMIN:

https://github.com/user-attachments/assets/c6578c3d-32a2-4746-86b0-8de00c299779

## Release Notes

- **New Features**
	- Introduced a `ReviewForm` for submitting scores and comments on applications.
	- Added a `ReviewScores` component to display average scores and individual reviews.
	- Implemented a new column for `reviewCount` in the applications table for better visibility.
	- Enhanced the `UpdateStatusDropdown` to restrict non-admin users from updating statuses.

- **Bug Fixes**
	- Improved error handling for score submissions in the `ReviewForm`.

- **Improvements**
	- Streamlined application data handling and display for a more efficient review process.

### CODE REVIEW

**Comment by owengretzinger** [comment]
@coderabbitai review

File: src/server/router/reviewers.ts
[the remark this answers is not in the document]
Comment (owengretzinger) [reply src/server/router/reviewers.ts:1]: Moved status to DH11Application table

File: src/server/router/reviewers.ts [hunk 1]
```diff
@@ -54,15 +63,40 @@ export const reviewerRouter = router({
...
+      }));
+
+      return applicationsWithReviewCount;
     }),
+
   getApplication: protectedProcedure
     .input(
       z.object({
         dh11ApplicationId: z.string().optional(),
+        includeHasReviewed: z.boolean().optional(),
```
[the remark this answers is not in the document]
Comment (owengretzinger) [reply src/server/router/reviewers.ts]: Removed the option and just always included reviews

**Review by arian81 — APPROVED** [review]
LGTM

**Review by Krish120003 — APPROVED** [review]

**Pull request merged**